components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowManagementServiceImpl.java

```diff
@@ -149,6 +205,10 @@ public WorkflowEvent getEvent(String id) {
                 }
                 event.setParameters(parameters);
             }
+
+            for (WorkflowListener workflowListener : workflowListenerList) {
+                workflowListener.doPostGetEvent(id);
+            }
```

Comment:
Take the listener execution outside the condition block. You may have to assign the result to a local variable and return later.